msal4j-sdk/src/test/java/com/microsoft/aad/msal4j/TestHelper.java

```diff
@@ -27,4 +38,21 @@ static void deleteFileContent(Class<?> classInstance, String resource)
         fileWriter.write("");
         fileWriter.close();
     }
+
+    static String generateToken() {
```

Comment:
This is fine. Other MSALs just create a json to represent the JWK. MSALs do not validate headers nor signatures, so only the payload matters.

You don't really need a "proper" token for OBO. MSAL never looks into it. 

You will need a proper id token though.

Reply:
In the latest commit I changed the name of the field that uses this to be more clear, but this is just used to create an assertion specifically for the OBO tests.

MSAL doesn't validate it, but internally we have some OAuth libraries that use the assertion to create an authorization grant, and as part of that it does some parsing, checks that it's signed, etc., and it was just quicker to create a 'real' one using that same library than to try to fake it or mock that part of MSAL.

Reply:
Ok, thanks for confirming. It may be worth adding a comment around this in code, as it's quite surprising.

Reply:
Yeah, here's an example where parsing happens: https://github.com/AzureAD/microsoft-authentication-library-for-java/blob/95b5efc9fc30514719e5b678c739351ddcf35fba/msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/OnBehalfOfRequest.java#L36

It's some very old cold (that line is from 2019, so pre-1.0), and seems like the AuthorizationGrant is used to make query parameters (?). I think reducing our usage of that package was something that was talked about years ago, and it might be worth looking into again.